### PR TITLE
chore: fix clippy warnings for rust 1.49

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Run `cargo +nightly clippy --all` and fix all the warnings. The Rust 1.49 update is scheduled for December 31st, but this PR can be merged now.

These are the new lints fixed by this PR:

https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_match
https://rust-lang.github.io/rust-clippy/master/index.html#comparison_to_empty
https://rust-lang.github.io/rust-clippy/master/index.html#field_reassign_with_default
https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains
https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro
https://rust-lang.github.io/rust-clippy/master/index.html#result_unit_err
https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_wraps

### CODE REVIEW

File: bridges/ethereum/src/eth.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use crate::config::Config;
 use crate::multibimap::MultiBiMap;
```
Comment (lrubiorod) [inline bridges/ethereum/src/eth.rs:2]: fmt

File: node/src/actors/chain_manager/mod.rs [hunk 2]
```diff
@@ -1736,7 +1698,7 @@ impl ChainManager {
             // Return all the blocks from this epoch range
             let hashes: Vec<(Epoch, Hash)> = block_chain_range.collect();
 
-            Ok(hashes)
+            hashes
```
Comment (lrubiorod) [inline node/src/actors/chain_manager/mod.rs:1701]: Could you remove the let hashes? Or the collect forces you to use it?

File: node/src/actors/chain_manager/mod.rs [hunk 3]
```diff
@@ -1837,7 +1793,10 @@ impl ChainManager {
 
             actix::fut::ok(())
         })
-        .map_err(|e, _, _| log::error!("{:?}", e));
+        .map_err(|(), _, _| {
+            // Errors at this point should be impossible because we explicitly ignore them
```
Comment (lrubiorod) [inline node/src/actors/chain_manager/mod.rs:1797]: if it is impossible, should we put a panic?

Comment (tmpolaczyk) [reply node/src/actors/chain_manager/mod.rs:1794]: I think we can also remove that `map_err`, but I will put a panic so we can find any potential errors.

File: node/src/actors/chain_manager/mod.rs (hunk 2 above)
Comment (tmpolaczyk) [reply node/src/actors/chain_manager/mod.rs]: Let's try

**Review by aesedepece — APPROVED** [review]

**Comment by tmpolaczyk** [comment]
Disabled `field_reassign_with_default` lint because of false positives: https://github.com/rust-lang/rust-clippy/issues/6515

**Pull request merged**